Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BinaryVsTextPerfTest.cs

```
@@ -0,0 +1,342 @@
+            }
+        }
+
+        private ITrace FindQueryBackendMetrics(ITrace trace)
+        {
+            Queue<ITrace> queue = new Queue<ITrace>();
+            queue.Enqueue(trace);
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Data.ContainsKey("Query Metrics"))
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private ITrace FindQueryTransitMetrics(ITrace trace)
+        {
+            Queue<ITrace> queue = new Queue<ITrace>();
+            queue.Enqueue(trace);
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Data.ContainsKey("Client Side Request Stats") & node.Name == "Microsoft.Azure.Documents.ServerStoreModel Transport Request")
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private ITrace FindQueryClientMetrics(ITrace trace, string name)
+        {
+            Queue<ITrace> queue = new Queue<ITrace>();
+            queue.Enqueue(trace);
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Name == name)
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private double CalculateAverage(List<double> AvgList)
+        {
+            AvgList = AvgList.GetRange(this.WarmupRounds, AvgList.Count - this.WarmupRounds);
```

>AvgList.GetRange(this.WarmupRounds, AvgList.Count - this.WarmupRounds); [](http://example.com/codeflow?start=22&length=71)

Avoid mixing concerns: a method that calculates average should not silently ignore part of its input

---

Done